### PR TITLE
Migrate ip sanity check script to IMDSv2

### LINKED ISSUE #990: IMDSv2: ip test script
Migrate the `examples/terraform-remote-exec-example/files/get-public-ip.sh` script to use Instance Metadata Service version 2. 

[Additional context.](https://gruntwork.atlassian.net/browse/IAC-1847)

### PR DESCRIPTION
**Pull request by zackproser**
Fixes #990 

These changes utilize[ the recommended format for simple IMDSv2 calls, shown here.](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html).

### CODE REVIEW

File: examples/terraform-remote-exec-example/files/get-public-ip.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 
 # For example purposes, print the public IP address of this instance
```
Comment (zackproser) [inline examples/terraform-remote-exec-example/files/get-public-ip.sh:3]:
```suggestion
# For example purposes, print the public IP address of this instance. This example uses Instance Metadata Service version 2.
```

File: examples/terraform-remote-exec-example/files/get-public-ip.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
 
-# For example purposes, print the public IP address of this instance
-wget -q -O- http://169.254.169.254/latest/meta-data/public-ipv4
-
+# For example purposes, print the public IP address of this instance. This example uses Instance Metadata Service version 2.
+TOKEN=`curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"` \
```
Comment (bwhaley) [inline examples/terraform-remote-exec-example/files/get-public-ip.sh:4]: NIT
```suggestion
TOKEN=$(curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600") \
```

**Comment by zackproser** [comment]
@bwhaley This one should be ready for another look! :pray:

**Review by bwhaley — APPROVED** [review]
LGTM

**Comment by zackproser** [comment]
Thanks for the review! Going to merge this in now.

**Comment by zackproser** [comment]
Fixes https://github.com/gruntwork-io/company/issues/133

**Pull request merged**